src/diffusers/models/unet_2d_blocks.py

```diff
@@ -662,7 +676,7 @@ def forward(
             hidden_states = attn(
                 hidden_states,
                 encoder_hidden_states=encoder_hidden_states,
-                attention_mask=attention_mask,
+                attention_mask=attention_mask if encoder_hidden_states is None else encoder_attention_mask,
```

Comment:
I don't understand this, why not just do:
```suggestion
                attention_mask=attention_mask
                encoder_attention_mask=encoder_attention_mask,
```